processing/pcap_filter.py

```diff
@@ -108,22 +125,12 @@ def _should_filter(self, p):
                 if p.qdcount > 0 and isinstance(p.qd, DNSQR):
                     req_name = p.qd.qname[:-1]  # remove dot
 
-                # extract response
-                if p.ancount > 0 and isinstance(p.an, DNSRR):
-                    for i in range(p.ancount):
-                        an = p.an[i]
-                        rdata = an.rdata
-                        if rdata[-1] == ".":
-                            res_name = rdata[:-1]
-                        else:
-                            res_host = rdata
-
-                if req_name and req_name.lower() in self.dns_ignore_list or \
-                        _domain(req_name.lower()) in self.dns_ignore_list:
-                    if res_name:
-                        self.dns_ignore_list.add(res_name)
-                    if res_host:
-                        self.host_ignore_list.add(res_host)
+                # extract dns response
+                res_names, res_hosts = self._extract_dns_response(p)
+
+                if self._ignore_req_name(req_name):
```

Comment:
the name `_ignore_req_name` is misleading, since you're actually performing the "ignorging" process in the next 2 lines